src/main/java/cook/Parser.java

```diff
@@ -0,0 +1,65 @@
+package cook;
+
+import java.util.HashMap;
+
+import exceptions.InvalidCommandException;
+
+/**
+ * Parser class to format inputs from the user.
+ */
+public class Parser {
+    private HashMap<String, Integer> commandArgs;
+
+    /**
+     * Constructor for Parser class.
+     *
+     * @param commandArgs Stores commands and their expected number of arguments.
+     */
+    public Parser(HashMap<String, Integer> commandArgs) {
+        this.commandArgs = commandArgs;
+    }
+
+    /**
+     * Constructor for Parser class.
+     *
+     * @param input Input to format.
+     * @return HashMap of commands and their corresponding inputs.
+     * @throws InvalidCommandException If command is not recognized.
+     */
+    public HashMap<String, String> readInput(String input) throws InvalidCommandException {
```

Comment:
I like the use of a verb for this method `readInput`, improving the code quality. Good job!